### DISCUSSION TITLE
Reference Action in another module

### DISCUSSION THREAD

**Discussion by mikecasas** [oqtane/oqtane.framework#726] [General]
I am trying to use ActionLink, and you typically use it like
```<ActionLink Action="Edit" Parameters="@($"id=" + context.TicketId.ToString())" />```

Where the Edit action typically corresponds to a Edit.razor file in the current folder.

I am using a multiple module design(Cats and Dogs modules in one solution), so is there a way to reference the action of another module? If I am in the Cats Index.razor file, can I link to the Edit.razor of the Dogs module?

**Comment by markdav-is** [comment]
is ActionLink a built-in component?

**Reply by sbwalker** [reply]
Yes you can, however you need to know the ModuleId of the other module. It is fairly easy to get the ModuleId using the PageState.Modules collection and a Linq statement.

**Reply by mikecasas** [reply]
It seems ActionLink is highly coupled to the module id. Can we create another parameter or just use another solution.

```
  _url = EditUrl(Action, _parameters);
```

```
 public string EditUrl(string action, string parameters)
 {
    return EditUrl(ModuleState.ModuleId, action, parameters);
 }
```

**Reply by sbwalker** [reply]
@mikecasas there are already a bunch of additional overloads in ModuleBase:

        public string EditUrl(string action)
        {
            return EditUrl(ModuleState.ModuleId, action);
        }

        public string EditUrl(string action, string parameters)
        {
            return EditUrl(ModuleState.ModuleId, action, parameters);
        }

        public string EditUrl(int moduleId, string action)
        {
            return EditUrl(moduleId, action, "");
        }

        public string EditUrl(int moduleId, string action, string parameters)
        {
            return EditUrl(PageState.Page.Path, moduleId, action, parameters);
        }

        public string EditUrl(string path, int moduleid, string action, string parameters)
        {
            return Utilities.EditUrl(PageState.Alias.Path, path, moduleid, action, parameters);
        }

**Reply by sbwalker** [reply]
Yes, we could add another optional parameter to ActionLink to allow a developer to specify the ModuleId. However the IsAuthorized method would also need to be refactored as it relies on ModuleState

**Reply by mikecasas** [reply]
I see that, but how can I use it without changing the ActionLink in the framework? Do I need to create my own ActionLink and then use one of the other ModuleBase overloads?

**Reply by mikecasas** [reply]
@sbwalker Would you want that in the framework if I do the refactoring?

**Reply by sbwalker** [reply]
@mikecasas yes, it should be in the framework - feel free to submit as a PR.
